src/vcf/assembly_report_writer.cpp

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "vcf/assembly_report_writer.hpp"
+
+namespace ebi
+{
+  namespace vcf
+  {
+    void SummaryAssemblyReportWriter::write_mismatch(const vcf::VcfVariant &vcf_variant) 
+    {
+        match_stats.num_variants++;
+    }
+    void SummaryAssemblyReportWriter::write_match(const vcf::VcfVariant &vcf_variant) 
+    {
+        match_stats.num_variants++;
+        match_stats.num_matches++;
+    }
+    void SummaryAssemblyReportWriter::write_results() 
+    {
+    	BOOST_LOG_TRIVIAL(info) << "Number of matches: " << match_stats.num_matches << "/" << match_stats.num_variants;
+    	BOOST_LOG_TRIVIAL(info) << "Percentage of matches: " << (static_cast<double>(match_stats.num_matches) / match_stats.num_variants) * 100 << "%";
+    }
+    void SummaryAssemblyReportWriter::add_result(bool result, const vcf::VcfVariant &vcf_variant) 
+    {
+        if(result) {
+            write_match(vcf_variant);
+        } else {
+            write_mismatch(vcf_variant);
+        }
+    }
+
+	OdbAssemblyReportWriter::OdbAssemblyReportWriter(const std::string &db_name) : db_name(db_name), current_transaction_size{0},
+                                                          transaction_size{1000000}
+	{	
+		try {
```

Comment:
Even though this is a work in progress, remember to use spaces instead of tabs. I think it's faster and easier to do that if you configure your editor to use spaces, and tabs don't even appear.